### PR TITLE
convert Avatar to typescript

### PR DESCRIPTION
**Pull request by danethurber**
todo:

- [x] ~generate proptypes~ i need to evaluate the following options. this will come in another pr
  - https://github.com/gcanti/prop-types-ts: requires a whole new ts library/syntax
  - https://www.npmjs.com/package/babel-plugin-typescript-to-proptypes: will require using babel to build
- [x] fix build
- [x] ~fix storybook addons~ i hit some snags with the storybook addons. i'm going to open another pr than includes storybook updates and new addons

### CODE REVIEW

File: packages/avatar/.storybook/main.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+  addons: [
+    // '@pluralsight/ps-design-system-storybook-addon-center',
+    // '@pluralsight/ps-design-system-storybook-addon-theme'
+  ],
```
Comment (danethurber) [inline packages/avatar/.storybook/main.js:5]: i need to do some upgrades on storybook through the project

File: packages/avatar/package.json [hunk 2]
```diff
@@ -9,16 +9,18 @@
   "main": "dist/cjs/index.js",
   "module": "dist/esm/index.js",
   "scripts": {
-    "build": "run-s build:js build:cjs build:css",
-    "build:cjs": "babel --root-mode upward src --out-dir dist/cjs",
+    "TODO_build:watch": "",
```
Comment (danethurber) [inline packages/avatar/package.json]: oops. i'll get watch working again

File: packages/avatar/package.json [hunk 3]
```diff
@@ -9,16 +9,18 @@
...
+    "build": "run-s build:cjs build:esm build:css",
+    "build:cjs": "tsc --project tsconfig.build.json --module commonjs --target es5 --outDir dist/cjs",
     "build:css": "build-css --useGlamor -i dist/cjs/css/index.js",
-    "build:js": "cross-env ESM=true babel --root-mode upward src --out-dir dist/esm",
-    "build:watch": "npm run build:js -- --watch",
+    "build:esm": "tsc --project tsconfig.build.json --module es2015 --target es5 --outDir dist/esm",
```
Comment (danethurber) [inline packages/avatar/package.json:15]: no babel needed

File: packages/avatar/src/js/sha1.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
+/* eslint-disable */
+
+/** NOTE: code originally found here https://github.com/jbt/tiny-hashes/blob/master/sha1/sha1.js */
+
+export default function sha1(b) {
```
Comment (danethurber) [inline packages/avatar/src/js/sha1.js:5]: i put this in a js file...i did not feel like typing it 😆

File: packages/avatar/src/react/index.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+          {...styles.initials(props)}
+          aria-label={name}
+          style={{ backgroundColor: getColorByName(name) }}
+        >
+          {initials}
+        </div>
+      )}
+    </div>
+  )
+}) as IAvatar
```
Comment (danethurber) [inline packages/avatar/src/react/index.tsx:105]: 🤮

File: packages/navbar/src/react/__specs__/__snapshots__/storyshots.spec.js.snap [hunk 6]
```diff
@@ -795,6 +795,7 @@ exports[`Storyshots Navbar desktop 1`] = `
         >
           <div
             data-css-79vgce=""
+            size="xSmall"
```
Comment (danethurber) [inline packages/navbar/src/react/__specs__/__snapshots__/storyshots.spec.js.snap:798]: this is wrong. i need to strip the size prop from the avatar div. i'll fix

File: packages/avatar/src/react/index.tsx [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+  size?: ValueOf<typeof sizes>
+  src?: string
+}
+
+interface AvatarStatics {
+  sizes: typeof sizes
+  widths: typeof widths
+}
+
+interface IAvatar
```
Comment (danethurber) [inline packages/avatar/src/react/index.tsx:36]: use `AvatarComponent` for the name

File: packages/avatar/src/react/index.tsx (hunk 5 above)
Comment (jaketrent) [reply packages/avatar/src/react/index.tsx:105]: yum!

**Review by jaketrent — APPROVED** [review]

**Pull request merged**